internal/sync/lifecycle.go

```diff
@@ -160,33 +150,29 @@ func (l *lifecycleOnce) WhenRunning(ctx context.Context) error {
 // If Stop is called multiple times it will return the error
 // from the first time it was called.
 func (l *lifecycleOnce) Stop(f func() error) error {
-	if l.stopping.Swap(true) {
-		<-l.stopCh
+	if l.state.CAS(int32(Idle), int32(Stopped)) {
+		close(l.startCh)
+		close(l.stopCh)
 		return l.err
```

Comment:
l.err is always nil here, we should return nil directly instead, just for clarity.